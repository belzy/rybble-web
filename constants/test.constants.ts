
type TTest = {
  SNAPSHOT_STRINGS: Array<string>
};

/**
 * Message strings to be used with testing functions like 'describe' and 'it'.
 * 
 * @example 
 * it(SNAPSHOT_STRINGS[0], () => {
 * 
 * // Your Test...
 * 
 * })
 */
export const TEST_STRINGS: TTest = {
  SNAPSHOT_STRINGS: [
    'should match snapshot'
  ]

};