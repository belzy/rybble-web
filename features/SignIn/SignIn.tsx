import React from 'react';
import {
  Text
} from '@chakra-ui/react';
import { AuthPage } from '../../components';
import { Link } from '../../components';
import { SignInForm } from './SignInForm';

const SignIn = (): React.ReactElement => {

  return (
    <AuthPage>

      <Text 
        fontSize='sm' 
        color='GrayText'
        ml='auto'
      >Don't have an account yet?

        <Link 
          href='/register' 
          ml='16px' 
          color='brand.600'
          _hover={{ color: 'brand.500' }}
          fontWeight='500'
        >SIGN UP</Link>

      </Text>

      <SignInForm />

    </AuthPage>
  );
};

export default SignIn;