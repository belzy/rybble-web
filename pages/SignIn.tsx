import React from 'react';

const SignIn = () => {

  return (
    <>Sign in page</>
  );
};

export default SignIn;