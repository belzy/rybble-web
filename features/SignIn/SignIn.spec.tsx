import { render } from '@testing-library/react';
import SignIn from './SignIn';
import '@testing-library/jest-dom';
import { TEST_STRINGS } from '../../constants/test.constants';

const { SNAPSHOT_STRINGS } = TEST_STRINGS;

describe('features/SignIn', () => {

  it(SNAPSHOT_STRINGS[0], () => {

    const { container } = render(<SignIn />);

    expect(container).toMatchSnapshot();

  });

});