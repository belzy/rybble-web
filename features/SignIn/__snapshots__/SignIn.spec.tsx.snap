// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/SignIn should match snapshot 1`] = `<div />`;
